Azure/sonic-mgmt
# [sanity check]: Add mux simulator sanity check

Signed-off-by: Lawrence Lee <lawlee@microsoft.com>


### Description of PR


Summary:
Closes #2753 

### Type of change




### Approach
#### What is the motivation for this PR?
Ensure that the mux cable simulator is fully functional prior to each test run

#### How did you do it?
* Send packets from the ToRs to the PTF, check that the active ToR's packet is received and the standby ToR's is not
    * I leveraged the fact that the `ping` command automatically sends ARP requests when it cannot find the target address, and check for the receipt of these ARP requests on the PTF. The ping command itself is expected to fail.
* Send packets from the PTF to the ToRs, check that both ToR's received the packet
    * Since there are no packet testing utils installed on the ToRs, I sent GARP packets to the ToRs and checked their neighbor tables to verify receipt.

Note that due to the new sanity check structure, `check_simulator_read_side` had to be renamed to `get_simulator_read_side` to prevent pytest from looking for a non-existent fixture named `check_simulator_read_side` (this is only necessary because `checks.py` iemports this method).

#### How did you verify/test it?
* Run a sanity check on a dual ToR testbed, ensure that it is able to pass and is also able to detect failures
* Run a sanity check on a single ToR testbed, ensure the mux simulator check does not run
* Run a sanity check on a T1 testbed, ensure the mux simulator check does not run

#### Any platform specific information?

#### Supported testbed topology if it's a new test case?

### Documentation
